test/model/test_models.py

```diff
@@ -264,6 +285,9 @@ def test_repr(Estimator, hyperparameters):
     [
         simple_feedforward_estimator(),
         deepar_estimator(),
+        # TODO: Enable this test: Error:  assert <gluonts.model.predictor.RepresentableBlockPredictor object at
+        # TODO: 0x124701240> == <gluonts.model.predictor.RepresentableBlockPredictor object at 0x124632940>
+        # TODO: deep_factor_estimator(),
```

Comment:
I think the problem here is due to the fact that some some sub-blocks are created in a global name scope, i.e. not in the name scope of the DeepFactorTrainingNetwork (or the prediction one): https://github.com/awslabs/gluon-ts/blob/master/src/gluonts/model/deep_factor/_estimator.py#L79-L91. This results in blocks in the original vs deserialized predictor having parameters whose names are different. So the equality test fails. I think this is a common issue with injecting outside-built blocks into a block which is being created, which is what is happening here: https://github.com/awslabs/gluon-ts/blob/master/src/gluonts/model/deep_factor/_estimator.py#L122-L129

I'm checking if there's a quick fix for this.